Fix floating point argument conversion in bun:ffi

jjrv opened the pull request:
### What does this PR do?

Don't try to convert JS values to floating point when calling a function through FFI that expects a double. Instead, handle integer and floating point values in the generated C wrapper code.

See #8430

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [X] Code changes

### How did you verify your code works?

- [X] I included a test for the new code, or existing tests cover it
- [X] I ran my tests locally and they pass (`bun-debug test test-file-name.test`)

jjrv commented:
Still need to figure out why the other two tests are broken on my machine. However they're already broken before the code change in this PR. Based on the required `Makefile` changes, the tests may not have been run on Linux before?

jjrv commented:
Still need to figure out why the other two tests are broken on my machine. However they're already broken before the code change in this PR. Based on the required `Makefile` changes, the tests may not have been run on Linux before?

jjrv commented on test/js/bun/ffi/ffi.test.js:
```diff
@@ -599,7 +601,7 @@ function ffiRunner(fast) {
 
           for (let i = min; i <= max; i += inc) {
-            expect(bigint ? BigInt(fn(i)) : fn(i)).toBe(i);
+            // expect(bigint ? BigInt(fn(i)) : fn(i)).toBe(i);
```
This was oddly already failing...

paperdave replied:
this was already broken. i had a pr to fix a different issue with uint32 but it causes another failure. which is funny because this is the test i wrote that is now failing because of that pr.

i do a stupid trick for uint32 and the sign bit is the edge case

jjrv replied:
I think I fixed it now, but the whole test has become unstable and I suspect it's not because of the number conversions

paperdave commented:
one thing jarred mentioned to me about about the approach of doing the check in C is 'how does it impact performance?'

could you do a benchmark in a release build?

jjrv commented:
> one thing jarred mentioned to me about about the approach of doing the check in C is 'how does it impact performance?'

I added a branch in autogenerated C code and removed a branch from the JS function calling it. It shouldn't make it worse. Is there an existing benchmark or is there reasonable doubt that branches in C are worse?

jjrv commented on test/js/bun/ffi/ffi.test.js:
```diff
@@ -605,6 +607,27 @@ function ffiRunner(fast) {
 
+    describe("Floating point identities for all valid exponents", () => {
+      const cases = [
+        // { type: "float", min: -149, max: 128, fn: identity_float },
```
Okayyy, it also corrupts floats so this test fails.

jjrv commented:
Hmm, with the latest test it just crashes when there's too many `expect` calls :/

jjrv commented:
Well, I'm giving up for now due to lack of time. Can't get the tests to run, even though I'm pretty sure the double handling would have been improved. GDB reports:

```
✓ run ffi > FFI runner (fast int) > integer identities work for all possible values > uint64_t [568.95ms]
[jest] test("primitives")

Thread 1 "bun-debug" received signal SIGSEGV, Segmentation fault.
0x0000020000905953 in ?? ()
(gdb) bt
#0  0x0000020000905953 in ?? ()
#1  0x00007fffad408038 in ?? ()
#2  0x00007ffffffe4c30 in ?? ()
#3  0x000055555f005876 in js_trampoline_op_call ()
#4  0x0000000000000000 in ?? ()
```
Looks like an FFI call jumps to a null pointer? Looks like it'd take some work to find the cause.
